### ISSUE TITLE
Generate yearly summaries for State of the Environment Reporting

### ISSUE THREAD

**Issue by Tasilee** [AtlasOfLivingAustralia/spatial-service#75]
The ALA is in an ideal position to support the generation of Essential Biodiversity Variables and State of the Environment indicators. We can leverage the volume of occurrence data to produce new 'environmental' layers for each bio-summary value. As a start we could match what Albert Van Dijk has done with http://wenfo.org/ausenv/#/2000/Tree_cover/Region/Actual/States_and%20Territories/bar,options/-28.96/135.00/4/none/Roadmap/Opaque

Areas needed to support (1 value/area) are 

State/Territories (got), 
SLAs (got), 
LGAs (got,
Catchments (got - level 2), 
Bioregions (got), 
CAPAD (parks/reserves) (got), 
Ramsar wetlands (got).  

In effect, under any map layer (year by value), there is a table that minimally contains polygon name/id, value in polygon. E.g., an entry for Sorell (LGA) for 2005 is 0.013 for endemism. Values would be calculated each year on the last day of the year from (minimally) 2000 as a baseline to present. 

The preliminary summary values are as much 'state of knowledge' but are a practical start-

Maximum endemism (I think this will be more informtive than averages)
Maximum occurrence density
Maximum richness
Maximum Shannon diversity
Number of endemics
Number of invasives
Number of Threatened (etc).
Number of bird species

We will automatically support the current intersect web service but we may need to generate/maintain a table of areas by the above values?

**Comment by Tasilee** [comment]
From meeting with Albert VanDijk December 4, 2017

1.	Monthly summary layers (occurrence density, species richness, endemicity and Shannon diversity) still seen as useful but spotty spatially. If reported at IBRA 7 subregion scale however (which implies some level of ecosystem homogeneity), maybe good. Noted that endemicity is less influenced by roads and population density. Yearly DOI snapshot. Two ways at least of generating the table.
 
2.	Focus on spatial subset of areas identified by loss of tree cover. Data for this is a 2x2 contingency table e.g., - 
 
                                      2017
                           Forest             Not forest
           Forest      static               Cleared         
2016
     Not forest    revegetated    static
 
3.	Focus on deforested areas and intersect with endemism and/or richness data for the area (values in grid cells). Could standardize by the occurrence density values for same cells. Idea is loss of richness is place and area dependent. Loss in central Australia is less important than in high diverse/endemic areas? Raw occurrence data please.
 
4.	Produce monthly summary layers for invasives and threatened species subsets
 
5.	Annual summary layers using 2000 as a baseline (keep all)
a.	Endemism
b.	Species richness
c.	Number of 'invasives'
d.	Number of 'threatened'
e.	Number of birds
f.	Graph of years vs values above (across all grid cells)

Would be good to have results before mid-February 2018 for 2017 report.

**Comment by Tasilee** [comment]
Given the type of functionality needed, it seems useful to have a generic layer difference function. You specify layer A and layer B and a new layer C is produced as A-B. Under Tools: Layer difference?

**Comment by Tasilee** [comment]
Other than the 6 summary layers, the deforested area produces a report (which could apply to any compare areas A and B scenario). In this case, we focus on the species list of occurrences IN the deforested area and OUT of that area. The (exportable) table has a line for each species IN the (deforested, or any given) area and columns

Species name
#Occurences OUT of the area
#Occurences IN the area  (note IN+OUT=total spacial occurrences in ALA)
%difference (100*IN/OUT)
Species code (I=invasive, T=threatened, blank=no category)

-------down the track as a generic tool, we could also have:

If OUT was a selected contextual layer, we have a column for each class (#occurrences in class)
If OUT was an environmental layer, we have columns for minimum, maximum and average.

**Comment by Tasilee** [comment]
In discussion with Adam, the strategy will be to drop the monthly updates totally and replace them with the yearly equivalents (run in January) with the additions using UP TO December 31 2000 as a baseline

1. Add number of invasives, threatened and birds (latter as demo) to occurrences, richness, endemism and shannon diversity 
1. Layers up to just prior end of year are all re-run as occurrences on prior years could have been added
1. We must maintain layers for each year from 2000

We have generated a new issue for the A-B contextual layer function. This will generate a square table of transitions between layers (which don't have to be years) with entries of AREA (sqkm) that LINKS to the relevant polygons/s. We can then use these polygons (e.g., forest to non-forest with Compare areas.

There is an existing issue for A-B for environmental layers (which the annual summaries will be) that will aid SOE reporting.
